dashboard/src/features/projects/rate-limiting/settings/components/RunServiceLimitingForm/RunServiceLimitingForm.tsx

```diff
@@ -0,0 +1,195 @@
+        ],
+      });
+    }
+  }, [loading, enabledDefault, ports, form]);
+
+  if (loading) {
+    return (
+      <ActivityIndicator
+        delay={1000}
+        label="Loading rate limits..."
+        className="justify-center"
+      />
+    );
+  }
+
+  const {
+    register,
+    formState: { errors },
+    formState,
+    watch,
+  } = form;
+
+  const enabled = watch('enabled');
+
+  const handleSubmit = async (formValues: RunServiceLimitingFormValues) => {
+    const updateConfigPromise = updateRunServiceRateLimit({
+      variables: {
+        appID: currentProject?.id,
+        serviceID: serviceId,
+        config: {
+          ports: ports.map((port, index) => {
+            const rateLimit = formValues.ports[index];
+            return {
+              ...port,
+              rateLimit: enabled
+                ? {
+                    limit: rateLimit.limit,
+                    interval: `${rateLimit.interval}${rateLimit.intervalUnit}`,
+                  }
+                : null,
+            };
+          }),
+        },
+      },
+    });
+    await execPromiseWithErrorToast(
+      async () => {
+        await updateConfigPromise;
+        form.reset(formValues);
+
+        if (!isPlatform) {
+          openDialog({
+            title: 'Apply your changes',
+            component: <ApplyLocalSettingsDialog />,
+            props: {
+              PaperProps: {
+                className: 'max-w-2xl',
+              },
+            },
+          });
+        }
+      },
+      {
+        loadingMessage: 'Updating Run service rate limit settings...',
+        successMessage: 'Run service rate limit settings updated successfully',
+        errorMessage: 'Failed to update Run service rate limit settings',
+      },
+    );
+  };
+
+  return (
+    <FormProvider {...form}>
+      <Form
+        onSubmit={handleSubmit}
+        className="flex h-full flex-col overflow-hidden"
+      >
+        <SettingsContainer
+          title={title}
+          switchId="enabled"
+          showSwitch
+          slotProps={{
+            submitButton: {
+              disabled: !formState.isDirty || maintenanceActive,
+              loading: formState.isSubmitting,
+            },
+          }}
+          className="flex flex-col px-0"
+        >
+          <Divider />
+          {ports.map((port, index) => {
+            if (port?.type !== 'http' || !port?.publish) {
+              return null;
+            }
+
+            const fieldTitle = `${port.type} <-> ${port.port}`.toUpperCase();
+            const showDivider = index < ports.length - 1;
+            return (
+              <>
+                <RateLimitField
+                  title={fieldTitle}
+                  disabled={!enabled}
+                  key={`ports.${port.port}`}
+                  register={register}
+                  errors={errors.ports}
+                  id={`ports.${index}`}
+                />
+                {showDivider && <Divider />}
+              </>
```

Comment:
This will probably show a warning on the console - try to wrap it with an actual element like a <div> and put the `key` prop on it